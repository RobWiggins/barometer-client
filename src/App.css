* {
  box-sizing: border-box;
  font-family: 'Open Sans', 'Source Sans Pro', 'Lato', sans-serif;
  color: #eee0c3;
}

.container {
  background-color: #0f1b30;
  color: #eee0c3;
}

html {
  max-width: 1000px;
  min-width: 340px;
  background-color: #0f1b30;

}

.app-header {
  padding-top: 15px;
  padding-bottom: 1px;
  padding-left: 5px;
  padding-right: 5px;
  background-color: #0f1b30;
}

.title-letter-holder {
  margin: 0;
  padding: 20px;
}

.app-title {
  font-weight: 500;
  font-family: 'Cormorant', 'Lora', 'Montserrat', 'Roboto', 'serif';
  font-size: 60px;
  margin-top: 5px;
}

/* rainbow title colors */
.blue {
  color: #405de6;
} 

.blue-purple {
  color: #5851db;
} 

.purple {
  color: #833ab4;
} 

.purple-pink {
  color: #c13584;
} 

.pink {
  color: #e1306c;
} 

.red {
  color: #fd1d1d;
} 

.red-orange {
  color: #f56040;
} 

.orange {
  color: #f77737;
} 

.orange-yellow {
  color: #fcaf45;
} 

.tagline {
  color: #eee0c3;
  font-weight: 500;
  font-family: 'Cormorant', 'Lora', 'Montserrat', 'Roboto', 'serif';
  font-size: 35px;
  margin-top: 5px;
  margin-bottom: 15px;
}

main {
  text-align: center;
  margin: auto;
}

#landingPane p {
  padding-left: 10px;
  padding-right: 10px;
}

.feature-summary {
  padding-top: 10px;
}

#search-home-img {
  width: 100%;
  height: auto; 
}

body {
  text-align: center;
  margin-left: 15px;
  margin-right: 15px;
  /* background-color: snow; */
  min-height: 100vh;
  border-radius: 1%;
  /* padding-bottom: 15px; */
  min-width: 290px;
  /* what to do here with body background */
}

#searchArea {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-content: space-between;
  padding-top: 20px;
}

.search img {
  height: 1em;
} 

.search button {
  border-radius: 0px 20px 20px 0px;
  height: 32px;
  cursor: pointer;
}

button:focus {
  outline-width: 0;
  outline-color: transparent;
}

#searchError,
#searchError > p {
  background-color: #ff9800;
  color: white;
  font-weight: 1.5em;
}

#searchError {
  margin-top: 20px;
  font-size: 18px;
}

.hidden {
  visibility: hidden;
}

.pastSearch {
  padding-left: 10px;
  padding-right: 10px;
  background-color: rgba(221, 221, 221, .5);
  border: none;
  color: black;
  text-align: center;
  font-size: 16px;
  margin: 4px 2px;
  transition: 0.3s;
  cursor: pointer;
  border-radius: 5%;
}

.pastSearch:hover {
  color: white;
  background-color: rgba(29, 161, 242, .6);
}

#searchHistory {
  padding-bottom: 15px;
}

#searchHistory h3 {
  margin-top: 20px;
}

#searchTweets {
  border-radius: 20px 0px 0px 20px;
  border: rgba(221, 221, 221, 1) solid 1px;
  padding: 2px;
  width: 220px;
  text-indent: 10px;
  margin: 0 auto;
  -webkit-tap-highlight-color: transparent;
  height: 32px; 
}

input:focus {
  outline-width: 0;
  outline-color: transparent;
}

.chart {
  min-width: 200px;
}

#emotionChart {
  padding: 10px;
}

#listTitle {
  padding-bottom: 10px;
  padding-top: 5px;
  font-size: 25px;
}

#tweets > ul {
  display: flex;
  justify-content: center;
}

#tweets li {
  font-family: 'Source Sans Pro', 'Open Sans', 'Lato', sans-serif;
  border-left: solid lightgray 1px;
  border-right: solid lightgray 1px;
  border-top: solid lightgray 1px;
  background-color: rgba(160, 204, 231, 0.03);
  border-radius: 1%;
  padding: 10px;
  margin-left: 8px;
  text-align: left;
  line-height: 1.4em;
}

li:last-child { 
  border-bottom: solid lightgray 1px; 
}

ul {
  list-style: none;
  padding-left: none;
  padding-inline-start: 0px;
}

@media only screen and (min-width: 768px) {

  main {
    text-align: center;
    margin: auto;
  }

  html {
    min-width: 290px;
    max-width: none;
  }

  #grid-holder {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 10px;
    grid-auto-rows: minmax(100px, auto);
    grid-auto-columns: 50%;
    grid-gap: 40px;
  }

  #emotionChart {
    padding: 10px;
    grid-column: 2;
    grid-row: 1;
  }

  #sentimentChart {
    padding-top: 15px;
    grid-column: 2;
    grid-row: 2;
  }

  #tweet-list {
    grid-column: 1;
    grid-row: 1 / 3;
    max-height: 800px;
    overflow: auto;
  }
}
