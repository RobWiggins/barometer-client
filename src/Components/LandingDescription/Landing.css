* {
  box-sizing: border-box;
}

nav,
footer {
  padding: 1% 0;
  background-color: #aaa;
}

nav {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 10;
}

#app-title {
  padding: 20px 15px;
  padding-bottom: 5px;
  background-color: rgba(29, 161, 242, .05);
}

.landingRow {
  color: #eee0c3;
  padding-bottom: 15px;
  line-height: 1.5em;
}

#callAction {
  margin-top: 15px;
  margin-bottom: 15px;
  padding-bottom: 15px;
}

#hideLanding {
  margin-top: 5px;
  width: 270px;
  padding-left: 10px;
  padding-right: 10px;
  margin: 10px;
  background-color: gainsboro;
  border: 1px solid gray;
  color: black;
  text-align: center;
  font-size: 24px;
  font-weight: bold;
  margin-bottom: 20px;
  transition: 0.3s;
  cursor: pointer;
  border-radius: 20px;
}



#hideLanding:hover {
  color: white;
  background-color: rgba(29, 161, 242, .6);

}

p {
  margin: 0 auto;
  max-width: 650px;
}

p + p {
  margin-top: 15px;
}

#thesis {
  padding-top: 5px; 
}
