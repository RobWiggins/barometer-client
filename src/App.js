import React from 'react';
// import { Route, Link } from 'react-router-dom';
import config from './config';
import './App.css';
import Header from './Components/Header/Header.js';
import FormQuery from './Components/FormQuery/FormQuery';
import EmotionChart from './Components/EmotionChart/EmotionChart';
import TweetList from './Components/TweetList/TweetList';
import ExampleChart from './Components/ExampleChart/ExampleChart';
import SentimentChart from './Components/SentimentChart/SentimentChart';
import ExampleSentimentChart from './Components/ExampleSentimentChart/ExampleSentimentChart';
import SearchError from './Components/SearchError/SearchError';
import SearchHistory from './Components/SearchHistory/SearchHistory'

class App extends React.Component {

  state = {
    watsonEmotionResults: null,
    tweets: null,
    queries: null,
    currentQuery: null,
    isSearchDisabled: false, // TODO not used at moment
    hasError: false,
  };

  /* retrieve past query history for all users */
  componentDidMount() {
    fetch(config.API_ENDPOINT + '/queries/history')
    .then(response => {
      if (!response.ok) {
        throw new Error({ message: 'error with getting history'});
      }
      return response.json()
    }).then(data => {
        console.log('QUERY HISTORY ON MOUNT: ', data.queries)
        this.setState( {
          queries: data.queries
        });
    })
    .catch(err => console.log(err.message))
  }
  
  /* Query valid and submitted, add query to history */
  /** TODO im doing this wrong, needs prev props */
  componentDidUpdate() {
    const body = JSON.stringify( {
      query: this.state.currentQuery
    });
    const options = { 
      method: 'POST', 
      headers: { 'content-type': 'application/json' },
      body,
    };

    /** TODO im doing this wrong */
    fetch(config.API_ENDPOINT + '/queries/history', options)
      .then(response => {
        if (!response.ok) {
          throw new Error({ message: 'error with retrieving history'});
        }
        return response.json()
      })
      .then(data => console.log('inserted query to history: ', data))
      .catch(err => console.log(err.message))
  }

  // returns false and disables search function if search > 25ch
  handleSearch(searchQuery) {
    // maximum search query length, allows limiting of watson targets
    //   const queryMaxLength = 25;
    //   if (searchQuery.length > queryMaxLength && !this.state.isSearchDisabled) {
    //     this.setState({
    //       isSearchDisabled: true,
    //   });
    //  } else if (searchQuery.length <= queryMaxLength && this.state.isSearchDisabled) {
    //     this.setState({
    //       isSearchDisabled: false,
    //     });
    //   }
  }

  /* TODO come back and wire up functionally with twitter retrieveTweets() */
  handleSubmitQuery = query => {

    console.log(query);
    /* try to communicate with backend */
    fetch(config.API_ENDPOINT + `/tweets/queries/${query}`) // how to send body?
      .then(response => {
        if (!response.ok) {
          throw new Error({ message: 'something seems to have gone wrong' });
        }
        return response.json();
      })
      .then(data => {
        // console.log('FROM THE FRONT END!!!!!', data);
        console.log('data.watsonEmotionResults: ', data.watsonEmotionResults);
        this.setState({
          watsonEmotionResults: data.watsonEmotionResults,
          tweets: data.duplicatesFiltered,
          hasError: false,
          currentQuery: data.currentQuery
        });
        
      })
      .catch(error =>
        this.setState({
          hasError: true,
        })
      ); // fix error message handling
  };

  render() {
    let isEmotionDataPresent = this.state.watsonEmotionResults ? true : false;
    let isTweetDataPresent = this.state.tweets ? true : false;
    console.log('isEmotionDataPresent: ', isEmotionDataPresent);

    let emotionChartDisplay, sentimentChartDisplay;
    if (isEmotionDataPresent) {
      emotionChartDisplay = (
        <EmotionChart watsonEmotionResults={this.state.watsonEmotionResults} />
      );
      sentimentChartDisplay = (
        <SentimentChart
          watsonEmotionResults={this.state.watsonEmotionResults}
        />
      );
    } else {
      emotionChartDisplay = <ExampleChart />;
      sentimentChartDisplay = <ExampleSentimentChart />;
    }

    let errorDisplay;
    if (this.state.hasError) {
      errorDisplay = <SearchError />;
    } else {
      errorDisplay = '';
    }

    return (
      <main className="main">
        <Header />
        <FormQuery
          isSearchDisabled={this.state.isSearchDisabled}
          handleSearch={this.handleSearch}
          handleSubmitQuery={this.handleSubmitQuery}
        />
        {errorDisplay}
        <SearchHistory queries={this.state.queries}></SearchHistory>
        <div id="grid-holder">
            {emotionChartDisplay}
            {sentimentChartDisplay}
            <TweetList tweets={this.state.tweets} />
        </div>
      </main>
    );
  }
}

export default App;
