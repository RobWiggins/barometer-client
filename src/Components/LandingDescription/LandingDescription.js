import React from 'react';
import './Landing.css'

export default class LandingDescription extends React.Component {
  render() {
    return (
      <div id="landingPane">
        <section className="landingRow">
          <header>
            <h3>
              Determine which emotions people on Twitter are feeling regarding
              any subject in real-time.
            </h3>
          </header>
          <p>
            <em>
              placeholder for screenshot of natural language emotion summary per
              query.
            </em>
          </p>
          <p>
            The Barometer web application grants users the ability to
            quantitatively measure the general population’s emotional view of
            events and subjects in real-time.
          </p>
            <p> Through combining Twitter's tweet
            search result API and IBM’s natural language processing model, the
            results are aggregated into 6 defined emotional sentiment categories
            (%): sadness, joy, fear, disgust, anger, and overall positive vs.
            negative outlook.
            </p> 
            <p id="thesis">The application is effectively a continuous,
            user-search driven online social presence survey.
            </p> 
        </section>
        <section id="callAction">
          <header>
            <h3>
              Record and quantify the online users' emotional viewpoint on any
              given subject.
            </h3>
          </header>
          <p>
            <em>
              placeholder for screenshot of emotion chart regarding subject
              matter.
            </em>
          </p>
          <p>
            Interactive charts and comparative displays enable easily digestible
            proprietary insights.
          </p>
        </section>
        <button id="hideLanding" onClick={() => this.props.onLandingButtonClick()}>Start searching now.</button>
      </div>
    );
  }
}
