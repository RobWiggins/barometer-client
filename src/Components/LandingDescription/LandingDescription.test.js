import React from 'react';
import ReactDOM from 'react-dom';
import LandingDescription from './LandingDescription';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../../App';

describe('testing LandingDescription Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <LandingDescription onLandingButtonClick={App.onLandingButtonClick} />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders a LandingDescription component snapshot', () => {
    const wrapper = shallow(
      <LandingDescription onLandingButtonClick={App.onLandingButtonClick} />
    ).find('#landingPane');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
