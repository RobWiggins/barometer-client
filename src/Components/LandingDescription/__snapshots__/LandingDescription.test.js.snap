// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing LandingDescription Component renders a LandingDescription component snapshot 1`] = `
<div
  id="landingPane"
>
  <section
    className="landingRow"
  >
    <header
      role="banner"
    >
      <h3>
        Determine which emotions people on Twitter are feeling regarding any subject in real-time.
      </h3>
    </header>
    <p>
      The Barometer web application grants users the ability to quantitatively measure the general population’s emotional view of events and subjects on Twitter.
    </p>
    <p>
       
      Through combining Twitter's tweet search result API and IBM’s natural language processing model, the results are aggregated into 6 emotional sentiment categories (%): sadness, joy, fear, disgust, anger, and overall positive vs. negative outlook.
    </p>
    <p>
      <img
        alt="Search query 'leonard clippers trade' with tweet results and emotion target keywords chart, indicating dominating sadness and joy emotional composition"
        id="search-home-img"
        src="./static/search_home.png"
      />
    </p>
    <p
      id="thesis"
    >
      The application is a continuous, user-search driven online social presence survey.
    </p>
  </section>
  <section
    id="callAction"
  >
    <header
      role="banner"
    >
      <h3>
        Record and quantify the online users' emotional viewpoint on any given subject.
      </h3>
    </header>
    <p>
      Interactive charts and comparative displays enable digestible proprietary insights.
    </p>
  </section>
  <button
    id="hideLanding"
    onClick={[Function]}
  >
    Start searching now.
  </button>
</div>
`;
